src/plot_recipes.jl

```diff
@@ -0,0 +1,195 @@
+using RecipesBase
+using DiffEqBase
+import .Trixi
+
+
+struct ContourPlot{Coordinates, Data, Variables, Limits, Vertices}
+  x::Coordinates
+  y::Coordinates
+  data::Data
+  variables::Variables
+  xlims::Limits
+  ylims::Limits
+  vertices_x::Vertices
+  vertices_y::Vertices
+end
+
+function ContourPlot(sol::ODESolution;
```

Comment:
This restriction a a bit arbitrary, I think. `ContourPlot` won't work for general `ODESolution`s (since we need Trixi's `semi` etc. and make a lot of corresponding assumptions).

Reply:
I agree! Should we choose to further pursue this general approach at all, I'd like to make the default to only supply an `u` (or `u_ode`) and a `semi`, since this is really all that's needed. Then, one (maybe) could have something like
```julia
function ContourPlot(sol::ODESolution{T, N, uType, uType2, DType, tType, rateType, P}) where 
    {T, N, uType, uType2, DType, tType, rateType,
     P<:ODEProblem{uType_, tType, isinplace, P_} where{uType_, tType, isinplace, P_<:AbstractSemidiscretization}}
  ContourPlot(sol.u[end], sol.prob.b)
end
```